test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -221,7 +232,101 @@ private void checkStreamDelete() {
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<String> reader =  clientFactory.createReader("reader-1", readerGroupId, new UTF8StringSerializer(), ReaderConfig.builder().build());
+        // It should not read an event from the stream because transaction was aborted.
+        EventRead<String>  eventRead =  reader.readNextEvent(READER_TIMEOUT_MS);
+        assertTrue(eventRead.getEvent() == null);
+        assertEquals("ABORTED", txn.checkStatus().toString());
+    }
+
+    /**
+     * This method tests the ability to successfully commit a transaction, including writing and reading events from a stream.
+     * @throws TxnFailedException if unable to commit or write to the transaction.
+     */
+    @Test(timeout = 20000)
+    private void checkTransactionReadAndWrite() throws TxnFailedException {
+        String scopeName = "transaction-test-scope";
+        String streamName = "transaction-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        TransactionalEventStreamWriter<String> writerTxn = clientFactory.createTransactionalEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+        assertNotNull(writerTxn);
+        // Begin a transaction.
+        Transaction<String> txn = writerTxn.beginTxn();
+        assertNotNull(txn.getTxnId());
+        int writeCount = 0;
+        // Writing 10 Events to the transaction
+        for (int event = 1; event <= 10; event++) {
+            txn.writeEvent("event test" + event);
+            writeCount++;
+        }
+        // Checking Status of transaction.
+        Transaction.Status status = txn.checkStatus();
+        assertEquals("OPEN", status.toString());
+        // Committing the transaction.
+        txn.commit();
+
+        String readerGroupId = UUID.randomUUID().toString().replace("-", "");
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build();
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<String> reader =  clientFactory.createReader("reader-1", readerGroupId, new UTF8StringSerializer(), ReaderConfig.builder().build());
+        EventRead<String> event = reader.readNextEvent(READER_TIMEOUT_MS);
+        int eventNumber = 1;
+        int readCount = 0;
```

Comment:
Instead of using two variable, can we do this using one only? i.e
` assertEquals(readCount, writeCount); ->  assertEquals(eventNumber-1, writeCount);`

Reply:
yeah removed extra variables.

Reply:
Still both eventNumber and readerCount is there.